### PR TITLE
Pract.07: References Add 07 | Daniel Jordanov

### PR DESCRIPTION
**Pull request by danielgo26**
Daniel Jordanov
6 3MI0600149

### CODE REVIEW

File: SI/Sem.07/Pract.07/References/Task07.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+int* SumTime(int hoursBegin, int minutesBegin, int hoursDuration, int minutesDuration)
+{
+	int endTime[2];
+	int hoursEnd = hoursBegin + hoursDuration;
+	int minutesEnd = minutesBegin + minutesDuration;
+
+	endTime[0] = hoursEnd;
+	endTime[1] = minutesEnd;
+	return endTime;
+}
```
Comment (GeorgiTerziev02) [inline SI/Sem.07/Pract.07/References/Task07.cpp:102]: What if minutesBegin + minutesDuration >= 60? Also this function again won't work as you expect it to work. The reason is the same;

File: SI/Sem.07/Pract.07/References/Task07.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	int maxLatestMinutes = -1;
+
+	for (int i = 0; i < countRows; i++)
+	{
+		int duration = movies[i][2];
+		int hoursOfDuration = ConvertToMinutes(duration)[0];
+		int minutesOfDuration = ConvertToMinutes(duration)[1];
+
+		int endHour = SumTime(movies[i][0], movies[i][1], hoursOfDuration, minutesOfDuration)[0];
+		int endMinutes = SumTime(movies[i][0], movies[i][1], hoursOfDuration, minutesOfDuration)[1];
```
Comment (GeorgiTerziev02) [inline SI/Sem.07/Pract.07/References/Task07.cpp:70]: @Justsvetoslavov Why these two lines work correctly?

Comment (GeorgiTerziev02) [reply SI/Sem.07/Pract.07/References/Task07.cpp:70]: + you are calculating the same thing two times

File: SI/Sem.07/Pract.07/References/Task07.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	endTime[0] = hoursEnd;
+	endTime[1] = minutesEnd;
+	return endTime;
+}
+
+void Print(int hoursEnd, int minutesEnd)
+{
+	if (hoursEnd >= 24)
+	{
+		hoursEnd -= (hoursEnd / 24) * 24;
```
Comment (GeorgiTerziev02) [inline SI/Sem.07/Pract.07/References/Task07.cpp:108]: %= 24

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

File: SI/Sem.07/Pract.07/References/Task07.cpp (hunk 2 above)
Comment (Justsvetoslavov) [reply SI/Sem.07/Pract.07/References/Task07.cpp]: @GeorgiTerziev02 We have reworked the solution in class.

As for the above question, it is returning a pointer and we are referencing by offset afterwards.(really hacky which we removed)

**Review by Justsvetoslavov — APPROVED** [review]

**Pull request closed without merging**